docs/understand/cmake_packages.rst

```diff
@@ -1,139 +1,188 @@
-
-Which uses the device compiler provided from the binary packages from
-`repo.radeon.com <http://repo.radeon.com>`_.
+    cmake_minimum_required(VERSION 3.8) # cxx_std_11
+    cmake_policy(VERSION 3.8...3.27)
+    project(MyProj LANGUAGES CXX)
+    find_package(hip REQUIRED)
+    add_library(MyLib ...)
+    target_link_libraries(MyLib PRIVATE hip::device)
+    target_compile_features(MyLib PRIVATE cxx_std_11)
 
 .. note::
-    Compiling for the GPU device requires at least C++11. This can be
-    enabled by setting ``CMAKE_CXX_STANDARD`` or setting the correct compiler flags
-    in the CMake toolchain.
-
-The GPU device code can be built for different GPU architectures by
-setting the ``GPU_TARGETS`` variable. By default, this is set to all the
-currently supported architectures for AMD ROCm. It can be set by passing
-the flag during configuration with ``-DGPU_TARGETS=gfx900``. It can also be
-set in the CMakeLists.txt as a cached variable before calling
-``find_package(hip)``::
-
-    # Set the GPU to compile for
-    set(GPU_TARGETS "gfx900" CACHE STRING "GPU targets to compile for")
-    # Search for rocm in common locations
-    list(APPEND CMAKE_PREFIX_PATH /opt/rocm/hip /opt/rocm)
-    # Find hip
-    find_package(hip)
-
-Using ROCm Libraries
----------------------------
-
-Libraries such as rocBLAS, MIOpen, and others support CMake users as
-well.
-
-As illustrated in the example below, to use MIOpen from CMake, you can
-call ``find_package(miopen)``, which provides the ``MIOpen`` CMake target. This
-can be linked with ``target_link_libraries``::
-
-    # Search for rocm in common locations
-    list(APPEND CMAKE_PREFIX_PATH /opt/rocm/hip /opt/rocm)
-    # Find miopen
-    find_package(miopen)
-    # Create library
-    add_library(myLib ...)
-    # Link with miopen
-    target_link_libraries(myLib MIOpen)
+    Compiling for the GPU device requires at least C++11.
 
-.. note::
-    Most libraries are designed as host-only API, so using a GPU device
-    compiler is not necessary for downstream projects unless it uses the GPU
-    device code.
+This project can then be configured with for eg.
+
+-  Windows: ``cmake -D CMAKE_CXX_COMPILER:PATH=${env:HIP_PATH}\bin\clang++.exe``
+
+-  Linux: ``cmake -D CMAKE_CXX_COMPILER:PATH=/opt/rocm/bin/amdclang++``
+
+Which use the device compiler provided from the binary packages of
+`ROCm HIP SDK <https://www.amd.com/en/graphics/servers-solutions-rocm>`_ and
+`repo.radeon.com <https://repo.radeon.com>`_ respectively.
 
+When using the CXX language support to compile HIP device code, selecting the
+target GPU architectures is done via setting the ``GPU_TARGETS`` variable.
+``CMAKE_HIP_ARCHITECTURES`` only exists when the HIP language is enabled. By
+default, this is set to some subset of the currently supported architectures of
+AMD ROCm. It can be set for eg. like ``-D GPU_TARGETS="gfx1032;gfx1035"``.
```

Comment:
it can be set to e.g., ``-D GPU_TARGETS="gfx1032;gfx1035"``

Reply:
Fixed.